Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs

```diff
@@ -396,6 +396,76 @@ public void EmbeddedObject_WhenAdditiveExplicit_ShouldThrow()
             });
         }
 
+        [Test]
+        public void WriteCopy_LocalToSync([Values(true, false)] bool originalEncrypted,
+                                          [Values(true, false)] bool copyEncrypted)
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var originalConfig = new RealmConfiguration(Guid.NewGuid().ToString())
+                {
+                    Schema = new[] { typeof(ObjectIdPrimaryKeyWithValueObject) }
+                };
+                if (originalEncrypted)
+                {
+                    originalConfig.EncryptionKey = TestHelpers.GetEncryptionKey(42);
+                }
+
+                var copyConfig = await GetIntegrationConfigAsync(Guid.NewGuid().ToString());
+                if (copyEncrypted)
+                {
+                    copyConfig.EncryptionKey = TestHelpers.GetEncryptionKey(23);
```

Comment:
When the copy is encrypted, we should test that we can't open it with a different encryption key.